### PR TITLE
Hierarchy view like in Vivado

### PR DESCRIPTION
**Pull request by rskvprd**
Just made hierarchy view more like in Vivado.

Maybe this is not production ready because I don't test any boundary cases :)

Screenshot 2025-04-01 at 02 21 14

### CODE REVIEW

**Comment by AndrewNolte** [comment]
Thanks for the contribution! Could you add these behind configs? Maybe showFileName and showChildCount. See the other config objects in that file for examples, then run "Extdev: update config" to update the package.json with those configs.

**Comment by rskvprd** [comment]
@AndrewNolte, I have added showFileName and showChildCount config objects to verilog.project.*. Is this correct place for these configs?

File: src/sidebar/ProjectComponent.ts [hunk 1]
```diff
@@ -103,6 +104,17 @@ export class InstanceItem extends ScopeItem {
     let item = await super.getTreeItem()
     item.contextValue = 'Module'
     item.iconPath = new vscode.ThemeIcon('chip')
+    if (this.definition && ext.project.showFileName.getValue()) {
+      let fileBasename = this.definition.doc.fileName.split('/').at(-1)
```
Comment (AndrewNolte) [inline src/sidebar/ProjectComponent.ts:108]: This won't work on windows

**Comment by AndrewNolte** [comment]
Can you change `showFileName`'s default to false? It's quite redundant for this project since modules are found by assuming that file names match the module name. I'll probably add this to the upcoming slang server extension, but only when the module name doesn't match the file name

File: src/sidebar/ProjectComponent.ts [hunk 2]
```diff
@@ -103,6 +104,17 @@ export class InstanceItem extends ScopeItem {
...
+      item.description = '(' + fileBasename + ')'
+    }
+    if (this.definition && this.hasChildren() && ext.project.showChildCount.getValue()) {
+      let instNum = this.definition.children.filter(
+        (child) => child.type === 'instance').length
```
Comment (AndrewNolte) [inline src/sidebar/ProjectComponent.ts:113]: Is this how vivado does it, only instances? I think it makes more sense to count the children in general, so this would depend on the filters

Comment (rskvprd) [reply src/sidebar/ProjectComponent.ts:113]: Yes, I think so (95% sure). Do you mean that count of variables, interfaces and so on will make more sense in a hierarchy view?

Comment (AndrewNolte) [reply src/sidebar/ProjectComponent.ts:113]: I think it's fine as is actually, let's see if others suggest otherwise
